### PR TITLE
ServiceBus Namespace: polling for the deletion

### PR DESCRIPTION
**Pull request by lrxtom2**
Deleting service bus in tanscation that resource group deletion will be blocked when sku set to premium.
Added WaitForCompletionRef method to make sure service bus was deleted and also added test for premium sku.

Thanks,
Chang

### CODE REVIEW

File: azurerm/resource_arm_servicebus_namespace.go [hunk 1]
```diff
@@ -212,10 +213,19 @@ func resourceArmServiceBusNamespaceDelete(d *schema.ResourceData, meta interface
 	resourceGroup := id.ResourceGroup
 	name := id.Path["namespaces"]
 
-	_, err = client.Delete(ctx, resourceGroup, name)
+	deleteFuture, err := client.Delete(ctx, resourceGroup, name)
```
Comment (katbyte) [inline azurerm/resource_arm_servicebus_namespace.go]: could we call this `future` to match the rest of the codebase?

File: azurerm/resource_arm_servicebus_namespace.go [hunk 2]
```diff
@@ -212,10 +213,19 @@ func resourceArmServiceBusNamespaceDelete(d *schema.ResourceData, meta interface
...
 	if err != nil {
 		return err
 	}
 
+	err = deleteFuture.WaitForCompletionRef(ctx, client.Client)
+	if err != nil {
```
Comment (katbyte) [inline azurerm/resource_arm_servicebus_namespace.go]: These lines could be combines as such:
```go
if err := deleteFuture.WaitForCompletionRef(ctx, client.Client); err != nil {
```

File: azurerm/resource_arm_servicebus_namespace_test.go [hunk 3]
```diff
@@ -221,3 +241,19 @@ resource "azurerm_servicebus_namespace" "test" {
 }
 `, rInt, location, rInt)
 }
+
+func testAccAzureRMServiceBusNamespace_premium(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+    name = "acctestRG-%d"
```
Comment (katbyte) [inline azurerm/resource_arm_servicebus_namespace_test.go]: Could we align these assignments?

File: azurerm/resource_arm_servicebus_namespace_test.go [hunk 4]
```diff
@@ -221,3 +241,19 @@ resource "azurerm_servicebus_namespace" "test" {
...
+    location = "%s"
+}
+resource "azurerm_servicebus_namespace" "test" {
```
Comment (katbyte) [inline azurerm/resource_arm_servicebus_namespace_test.go]: And add a space here between the resources

**Comment by tombuildsstuff** [comment]
Tests pass:

screenshot 2018-09-12 at 17 18 05

**Pull request merged**